File: app/src/main/java/com/olair/picsart/camera/utils/CameraExecutor.java
package com.olair.picsart.camera.utils;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

/**
 * Create by oLair on 2020/10/29.
 */
public enum CameraExecutor {

    BACKGROUND(Executors.newSingleThreadExecutor()),
    ;

    private Executor executor;

    CameraExecutor(Executor executor) {
        this.executor = executor;
    }

    public void post(Runnable runnable) {
        executor.execute(runnable);
    }
}

File: app/src/main/java/com/olair/picsart/camera/utils/CameraExecutor.kt
package com.olair.picsart.camera.utils

import java.util.concurrent.Executor
import java.util.concurrent.Executors

/**
 * Create by oLair on 2020/10/29.
 */
enum class CameraExecutor(private val executor: Executor) {
    BACKGROUND(Executors.newSingleThreadExecutor());

    fun post(runnable: Runnable) {
        executor.execute(runnable)
    }
}